### ISSUE TITLE
open new form in a separate application

### ISSUE THREAD

**Issue by MadsCastle7** [JJConsulting/JJMasterData#256]
I am using your FormView example to view a dictionary that i've created in a separate application. 

Do you have a method to open a from for a particular row in edit mode, or open a new form dynamically from a separate application?

**Comment by MadsCastle7** [comment]
I have worked out how to open a new blank form in a separate application. Do you have a method of returning the id of the row that has been inserted upon save?

**Comment by LucioPelinson** [comment]
Hello, @MadsCastle7 
Where and how do u like to return the generated Id? 
Would you have an example of what you are trying to do?

**Comment by MadsCastle7** [comment]
I would like the Id to be generated after an InsertAsync. By way of example, I may wish to complete a form about the details of a hospital bed. Once i have completed those details, I then need the returned Id in order that i can link the hospital bed to a patient. The patient information may have been taken from somewhere like an API and would not be anythnig to do with Dynamic Forms.

See attached 
![JJMaster](https://github.com/user-attachments/assets/58052bd3-c322-4519-afea-0cb738bcbc20)

**Comment by MadsCastle7** [comment]
i note that EntityProviderBase.cs goes someway to achieving this, but the Id is not returned. 
![image](https://github.com/user-attachments/assets/af6258ff-9023-4ab0-8336-ae277e6b6b85)

**Comment by MadsCastle7** [comment]
Hi Lucio, i've just noted that the Dictionary item is passed by reference and therefore we can access the new identity seed that has been added from there. I suppose it wouls be nice if the method returned the id though

**Comment by LucioPelinson** [comment]
@MadsCastle7, you can get the Id by reference.
After the insert command the id will be in the values hash.

Ex.:
```cs
// after insert command
var myId = values["id"];
```
If you are using procedures, check if after the insert command at procedure it's returning the id variable. 
<br>Example:
```sql
...
IF @TYPEACTION = 'I' 
	BEGIN 
		INSERT INTO [TB_TEST] (
			[name],
			[age])
		VALUES (
			@name,
			@age)
		SET @RET = 0; 
                 OUTPUT Inserted.id --THIS LINE IS IMPORTANT
	END 
...
```

**Comment by LucioPelinson** [comment]
> Hi Lucio, i've just noted that the Dictionary item is passed by reference and therefore we can access the new identity seed that has been added from there. I suppose it wouls be nice if the method returned the id though

Yes, we have been thinking about this for a long time, but this will generate a big breaking change.

**Comment by MadsCastle7** [comment]
thanks for the replies, in addition can you please tell me if its possible to create additional custom field component, for example, i notice there is no Check Box List Component

**Comment by MadsCastle7** [comment]
The logging code uses inline SQL, are there any plans for this to be replaced with Stored Procedures?

**Comment by LucioPelinson** [comment]
> The logging code uses inline SQL, are there any plans for this to be replaced with Stored Procedures?

Yes, it is in development at 4.3 version

**Comment by LucioPelinson** [comment]
> thanks for the replies, in addition can you please tell me if its possible to create additional custom field component, for example, i notice there is no Check Box List Component

No yet... any interfaces is missing<br>
But you can use a ComboBox Component and enable mult select option
![image](https://github.com/user-attachments/assets/a51e7bdf-6584-4328-89ef-28e6be07f393)

or<br>

Use a Relation
https://md.jjconsulting.tech/articles/tutorials/relationship.html

**Comment by MadsCastle7** [comment]
> > The logging code uses inline SQL, are there any plans for this to be replaced with Stored Procedures?
> 
> Yes, it is in development at 4.3 version

Morning Lucio, do you have timescales for version 4.3?

**Comment by LucioPelinson** [comment]
> > > The logging code uses inline SQL, are there any plans for this to be replaced with Stored Procedures?
> > 
> > 
> > Yes, it is in development at 4.3 version
> 
> Morning Lucio, do you have timescales for version 4.3?

The release of version 4.3 will be on 2024-08-28.

**Comment by gumbarros** [comment]
@MadsCastle7 any other question? I will close this issue.
